File: Magnets.py
n = int(input())
l = []
for i in range(n):
    a, b = list(input())
    l.append(a)
    l.append(b)

c = 1

for i in range(1, len(l)-1, 2):
    if l[i] == l[i+1]:
        c += 1

print(c)

File: Magnets.cpp
#include<bits/stdc++.h>

using namespace std;

int main()

{
    //freopen("input.txt","r",stdin);
    int n,a,b,c=1;
    cin >> n;
    cin >> a;
    for (int i = 1; i < n ; ++i)
    {
        cin >> b;
        if (a != b)
            c++;
        a = b;
    }

    cout << c << "\n";
    return 0;
}
